src/recipes.jl

```diff
@@ -719,24 +719,39 @@ end
 
     # create a list of shapes, where each shape is a single boxplot
     shapes = Shape[]
-    groupby = extractGroupArgs(d[:x])
-
-    for (i, glabel) in enumerate(groupby.groupLabels)
 
-        # get the edges and widths
-        y = d[:y][groupby.groupIds[i]]
-        widths, centers = violin_coords(y, trim=trim)
+    legend --> false
 
+    if length(d[:x])==1
+        widths, centers = violin_coords(d[:y], trim=trim)
         # normalize
         widths = _box_halfwidth * widths / maximum(widths)
 
         # make the violin
-        xcenter = discrete_value!(d[:subplot][:xaxis], glabel)[1]
+        xcenter = discrete_value!(d[:subplot][:xaxis], d[:x][1])[1]
```

Comment:
Mauro I can't work on this now, but I think we need to talk it through.
Maybe put this on hold until I have time?

On Friday, June 17, 2016, Mauro <notifications@github.com> wrote:

> In src/recipes.jl
> <https://github.com/tbreloff/Plots.jl/pull/339#discussion_r67479348>:
>
> >          # normalize
> >          widths = _box_halfwidth * widths / maximum(widths)
> >
> >          # make the violin
> > -        xcenter = discrete_value!(d[:subplot][:xaxis], glabel)[1]
> > +        xcenter = discrete_value!(d[:subplot][:xaxis], d[:x][1])[1]
>
> Not sure this is right.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tbreloff/Plots.jl/pull/339/files/f7c06da93e569b90cf88f9805185f25ed4dc60c2#r67479348>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AA492pTMJd-AMDogHfFn3UdW5f_i5V35ks5qMmIRgaJpZM4I4KUg>
> .
>